include/libdnf/base/transaction.hpp

```diff
@@ -86,14 +87,14 @@ class Transaction {
     /// @param callbacks Callbacks to be called during rpm transaction.
     /// @param description Description of the transaction (the console command for CLI,
     //                     verbose description for API usage)
-    /// @param user_id UID of the user that started the transaction.
     /// @param comment Any comment to store in the history database along with the transaction.
+    /// @param user_id UID of the user that started the transaction.
     /// @return An enum describing the result of running the transaction.
     TransactionRunResult run(
         std::unique_ptr<libdnf::rpm::TransactionCallbacks> && callbacks,
         const std::string & description,
-        const std::optional<uint32_t> user_id,
-        const std::optional<std::string> comment);
+        const std::string & comment = {},
```

Comment:
Here I don't see a difference between empty string and not providing a string, therefore I am fine with the approach. I am even fine if the approach here will be different with a solution for query filters.